firmware/Core/Src/mpi/mpi_command_handling.c

```diff
@@ -0,0 +1,118 @@
+#include "mpi/mpi_command_handling.h"
+#include "mpi/mpi_types.h"
+#include "debug_tools/debug_uart.h"
+#include "uart_handler/uart_handler.h"
+#include "main.h"
+#include "stm32l4xx_hal.h"
+#include "string.h"
+#include <stdio.h>
+
+enum Mpi_Mode current_mpi_mode = NOT_LISTENING_TO_MPI_MODE;
+
+/**
+ * @brief Sends commandcode+params to the MPI as bytes
+ * 
+ * @param bytes_to_send Buffer containing the CMD + Params (IF ANY) as hex bytes
+ * @param bytes_to_send_len Size of CMD buffer
+ * @param mpi_cmd_response Buffer to store incoming response from the MPI
+ * @param mpi_cmd_response_len Size of response buffer
+ * @return 0: Success, 1: Failed UART transmission, 2: Failed UART reception, 3: MPI failed to execute CMD
+ */
+uint8_t MPI_send_telecommand_hex(const uint8_t *bytes_to_send, size_t bytes_to_send_len, uint8_t mpi_response[], size_t mpi_response_size)
+{
+	// Set MPI to command mode
+	current_mpi_mode = COMMAND_MODE;
+
+	// Transmit Buffer to store CMD+Params formatted to be sent to the MPI
+	size_t UART1_txBuffer_size = bytes_to_send_len + 2;
+	//size_t UART1_rxBuffer_size = UART1_txBuffer_size + 1;
+	uint8_t UART1_txBuffer[UART1_txBuffer_size];	   	// +2 accounts for 'TC' command identifier being appended
```

Comment:
Can you just make it so that the `TC` part has to be included from the caller, and then send the buffer directly (instead of this copy step).

Valuable in case the command format to the MPI changes with a firmware upgrade, or in case we need to send other data to recover the MPI after a bad firmware upgrade, etc.